### PR TITLE
Install powershell and azure-cli on deb 12

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Comment by lbussell** [comment]
Related: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1185

File: src/debian/12/gcc14/amd64/Dockerfile [hunk 1]
```diff
@@ -29,3 +30,17 @@ ENV LANG=en_US.utf8
...
     ln -s $(command -v g++) /usr/bin/g++-14
+
+# Install powershell. This is in the debpkg/amd64 docker file as it may not be required in the arm64
+# images, and is not actually supported over there.
+RUN apt-get update && \
+    apt-get install -y \
+      apt-transport-https \
+      software-properties-common \
+      curl && \
+    curl -sL https://packages.microsoft.com/config/debian/11/packages-microsoft-prod.deb -o packages-microsoft-prod.deb && \
```
Comment (carlossanlop) [inline src/debian/12/gcc14/amd64/Dockerfile:41]: Shouldn't this be 12?
```suggestion
    curl -sL https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb -o packages-microsoft-prod.deb && \
```

Comment (carlossanlop) [reply src/debian/12/gcc14/amd64/Dockerfile:41]: Or is this due to what you mentioned in chat?
> the newest version of the azure-cli availble doesn't appear to have the correct parameters to support the AzureCLI task

**Review by carlossanlop — APPROVED** [review]
LGTM aside from my comment/question.

File: src/debian/12/gcc14/amd64/Dockerfile (hunk 1 above)
Comment (mmitche) [reply src/debian/12/gcc14/amd64/Dockerfile:41]: powershell isn't actually available on the debian 12 feed for whatever reason

Comment (mmitche) [reply src/debian/12/gcc14/amd64/Dockerfile:40]: Updated comment

**Pull request merged**